src/lib.rs

```diff
@@ -1778,18 +1825,32 @@ impl GooseAttack {
                         }
 
                         self.stats.requests.insert(key.to_string(), merge_request);
-                        message = parent_receiver.try_recv();
+                        message = request_stats_receiver.try_recv();
+                    }
+
+                    // If we're printing task statistics, collect the final messages received
+                    // from users.
+                    if !self.configuration.no_task_stats {
+                        // Load raw_task messages from user threads until the receiver queue is empty.
+                        let mut message = task_stats_receiver.try_recv();
+                        while message.is_ok() {
+                            let raw_task = message.unwrap();
+                            // Store a new statistic.
+                            self.stats.tasks[raw_task.taskset_index][raw_task.task_index]
+                                .set_time(raw_task.run_time, raw_task.success);
+                            message = task_stats_receiver.try_recv();
+                        }
                     }
                 }
 
                 #[cfg(feature = "gaggle")]
                 {
-                    // As worker, push request statistics up to manager.
+                    // As worker, push all statistics up to manager.
                     if self.configuration.worker {
-                        // Push request statistics to manager process.
                         worker::push_stats_to_manager(
                             &socket.clone().unwrap(),
-                            &self.stats.requests.clone(),
+                            Some(self.stats.requests.clone()),
+                            Some(self.stats.tasks.clone()),
                             true,
```

Comment:
This I like 👍

Reply:
Now that we're using an Enum, I've reworked instead to use a Vec() -- this will make it easier to add additional types of metrics in the future.